File: mediator/java/ChatRoomMediator.java
package Mediator;

import java.util.Dictionary;
import java.util.Hashtable;

// Mediator (IChatRoomMediator) arayüzünü uygular.
// Colleague (User) nesneleri arasındaki iletişimi koordine eder.
// UML diyagramındaki ConcreteMediator yapısına denk gelmektedir.
class ChatRoomMediator implements IChatRoomMediator {
    private Dictionary<Integer, User> _userDictionary;

    public ChatRoomMediator() {
        _userDictionary = new Hashtable<>();
    }

    public void addUserInRoom(User user) {
        _userDictionary.put(user.id, user);
    }

    public void sendMessage(String message, int userId) {
        User user = _userDictionary.get(userId);
        user.receiveMessage(message);
    }
}

File: mediator/kotlin/ChatRoomMediator.kt
package Mediator

import java.util.Dictionary
import java.util.Hashtable

// Mediator (IChatRoomMediator) arayüzünü uygular.
// Colleague (User) nesneleri arasındaki iletişimi koordine eder.
// UML diyagramındaki ConcreteMediator yapısına denk gelmektedir.
internal class ChatRoomMediator : IChatRoomMediator {
    private val _userDictionary: Dictionary<Int, User>

    init {
        _userDictionary = Hashtable()
    }

    override fun addUserInRoom(user: User) {
        _userDictionary.put(user.id, user)
    }

    override fun sendMessage(message: String, userId: Int) {
        val user = _userDictionary.get(userId)
        user.receiveMessage(message)
    }
}